src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -38,11 +38,11 @@
     </ResolveNuGetPackagesDependsOn>
   </PropertyGroup>
 
-  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
+  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson"/>
 
   <Target Name="GenerateAllTestProjectJsons" DependsOnTargets="GenerateTestProjectJson" />
 
-  <Target Name="DetermineProjectJsonPath">
+  <Target Name="DetermineProjectJsonPath" Condition="'$(UsesCommonTestProjectJson)'!='true'">
```

Comment:
I definitely don't like seeing this condition duplicated all over the place. You should use something along the lines of 

```
  <PropertyGroup>
    <GenerateTestProjectJsonDependsOn Condition="'$(SkipGenerateTestProjectJson)'!='true'">
      ExtractProjectReferences;
      GatherProjectReferenceForProjectJson;
      DetermineProjectJsonPath;
      AddDependenciesToProjectJson
    </GenerateTestProjectJsonDependsOn>
  </PropertyGroup>
 
  <Target Name="GenerateTestProjectJson" DependsOnTargets="$(GenerateTestProjectJsonDependsOn)" />
```